backend/battle/views.py

```diff
@@ -46,7 +46,14 @@ class TeamView(LoginRequiredMixin, CreateView):
     success_url = reverse_lazy("invite")
 
     def get_initial(self):
-        initial = {"battle": self.kwargs['pk'], "trainer": self.request.user.id}
+        battle_id = self.kwargs['pk']
+
+        try:
+            Battle.objects.get(id=battle_id)
```

Comment:
Nice, this can be streamlined by `get_object_or_404`

Reply:
If the battle doesn't exist we should return an error page. Some people discuss that the behavior should be different to avoid leaking information of the application but for this case let's return a 404

Reply:
Alright!